src/OperationProcessor.ts

```diff
@@ -233,10 +240,22 @@ class OperationProcessorImpl implements OperationProcessor {
       throw Error('Invalid operation: batchFileHash undefined');
     }
 
+    this.operationStore.store(opHash, operation);
+
+    const did = this.getDid(operation, opHash);
+    this.getDeferredOperationsList(did).append(opHash);
+
+    return opHash;
+  }
+
+  /**
+   * Processes a specified DID state changing operation.
+   */
+  private async processDeferred (operation: WriteOperation, opHash: OperationHash): Promise<void> {
     // opInfo is operation with derivable properties projected out
     const opTimestamp: OperationTimestamp = {
```

Comment:
Design: Should `OperationTimestamp` and `OperationInfo` be revisited? Since `this.opeartionStore` is readily available and is being used throughout already, maybe there is no point in copying/duplicating most of the same info like between line 255 ~ 266.

Perhaps `OperationInfo` can be evolved into just `OperationState`:

```javascript
interface OperationState {
  status: OperationStatus;
  missingAncestor?: VersionId;
}

```